examples/mnist_tfrecord.py

```diff
@@ -0,0 +1,108 @@
+'''MNIST dataset with TensorFlow TFRecords.
+
+Gets to 99.25% test accuracy after 12 epochs
+(there is still a lot of margin for parameter tuning).
+'''
+import os
+import copy
+import time
+
+import numpy as np
+
+import tensorflow as tf
+from keras import backend as K
+from keras.models import Model
+from keras import layers
+from keras import objectives
+from keras.utils import np_utils
+from keras import objectives
+
+from tensorflow.contrib.learn.python.learn.datasets.mnist import load_mnist
+
+if K.backend() != 'tensorflow':
+    raise RuntimeError('This example can only run with the '
+                       'TensorFlow backend for the time being, '
+                       'because it requires TFRecords, which '
+                       'are not supported on other platforms.')
+
+
+def cnn_layers(x_train_input):
+    x = layers.Conv2D(32, (3, 3),
+                      activation='relu', padding='valid')(x_train_input)
+    x = layers.Conv2D(64, (3, 3), activation='relu')(x)
+    x = layers.MaxPooling2D(pool_size=(2, 2))(x)
+    x = layers.Dropout(0.25)(x)
+    x = layers.Flatten()(x)
+    x = layers.Dense(128, activation='relu')(x)
+    x = layers.Dropout(0.5)(x)
+    x_train_out = layers.Dense(classes,
+                               activation='softmax',
+                               name='x_train_out')(x)
+    return x_train_out
+
+
+sess = tf.Session()
+K.set_session(sess)
+
+batch_size = 256
+batch_shape = [batch_size, 28, 28, 1]
+steps_per_epoch = 1000
```

Comment:
Maybe we should use the number of training samples in MNIST divided by the batch size (rounded up) here?